src/bin/inx-chronicle/stardust_inx/chunks.rs

```diff
@@ -0,0 +1,344 @@
+// Copyright 2022 IOTA Stiftung
```

Comment:
You maybe want to also port and adapt the unit tests from `itertools`? (also ... maybe consider giving tribute to the original authors - I haven't checked against their code though)

Reply:
There are no tests to adapt. And regarding crediting them, well, I don't think there's any reason to do that. This isn't a library 😆

Reply:
maybe it's just me who finds this code non-trivial :see_no_evil: